packages/react-router-dom/__tests__/data-browser-router-test.tsx

```diff
@@ -2895,7 +2973,7 @@ function testDomRouter(
           let { container } = render(<RouterProvider router={router} />);
 
           expect(container.querySelector("form")?.getAttribute("action")).toBe(
-            "/foo?a=1"
+            "/foo/bar?a=1"
```

Comment:
This existing test was asserting the buggy behavior.  When no `action` is specified it should inherit the current path including the splat portion